### PR TITLE
Implement Michigan Family Independence Program (FIP)

### LINKED ISSUE #7013: Michigan TANF

### PR DESCRIPTION
**Pull request by hua7450**
## Summary

Implements Michigan Family Independence Program (FIP), the state's TANF cash assistance program.

Closes #7013

---

### Regulatory Authority
- **Michigan DHHS BEM 518** - FIP/RCA/SDA Income Budgeting
- **Michigan DHHS BEM 520** - Computing the FIP/RCA/SDA Budget
- **Michigan DHHS BEM 400** - Assets
- **Michigan TANF State Plan 2023/2025**

---

#### Qualifying Deficit Test (New Applicants)
- **Applies to:** New applicants only
- **Deduction rate:** 20% of remaining earned income after flat $200 deduction
- **Test:** Countable income < Payment Standard

Source: [BEM 520 Section C (p.3)](https://mdhhs-pres-prod.michigan.gov/olmweb/EX/BP/Public/BEM/520.pdf#page=3)

#### Issuance Deficit Test (Enrolled Recipients)
- **Applies to:** Currently enrolled recipients
- **Deduction rate:** 50% of remaining earned income after flat $200 deduction
- **Test:** Countable income < Payment Standard

Source: [BEM 520 Section D (p.4)](https://mdhhs-pres-prod.michigan.gov/olmweb/EX/BP/Public/BEM/520.pdf#page=4)

---

#### A. Earned Income Flat Deduction

| Deduction Type | Amount |
|----------------|--------|
| Flat (per working person) | $200 |

**Critical:** Deduction applies **per person** - each worker in the household receives their own $200 deduction.

Source: [BEM 518 (p.5)](https://mdhhs-pres-prod.michigan.gov/olmweb/ex/BP/Public/BEM/518.pdf#page=5)

#### B. Earned Income Percentage Deduction

| Purpose | Percentage | Effective Date |
|---------|------------|----------------|
| Initial eligibility (qualifying) | 20% | 2008-10-01 |
| Benefit calculation (issuance) | 50% | 2011-01-01 |

**Applied to:** Remaining earned income after $200 flat deduction

Source: [BEM 518 (p.5)](https://mdhhs-pres-prod.michigan.gov/olmweb/ex/BP/Public/BEM/518.pdf#page=5)

---

### 3. INCOME STANDARDS BY FAMILY SIZE

| Family Size | Payment Standard (2017) | Payment Standard (2024) |
|-------------|------------------------|------------------------|
| 1 | $306 | $363 |
| 2 | $403 | $478 |
| 3 | $492 | $583 |
| 4 | $597 | $707 |
| 5 | $694 | $822 |
| 6 | $828 | $981 |
| 7 | $905 | $1,072 |
| 8+ | +$80/person | +$95/person |

Source: [Michigan TANF State Plan 2017 (p.7)](https://www.michigan.gov/-/media/Project/Websites/mdhhs/Folder1/Folder33/TANF_State_Plan_2017.pdf#page=7), [Reference Table 210](https://mdhhs-pres-prod.michigan.gov/olmweb/ex/RF/Public/RFT/210.pdf#page=1)

---

#### Formula:
```
Benefit = Payment Standard − Countable Income
```

Where **Countable Income** (per person) =
```
max(Gross Earned − $200 − 50% of Remainder, 0) + Gross Unearned
```

#### Calculation Steps (Per Person):
1. Determine gross earned income for each working person
2. Subtract $200 flat deduction from each person's earnings
3. Subtract 50% of remaining earned income (for benefit calculation)
4. Result cannot go below $0
5. Sum all persons' countable earned income
6. Add gross unearned income = **Total Countable Income**
7. Benefit = Payment Standard − Countable Income

#### Example: Two-Earner Household (Family of 4)
```
Parent 1: $800 earned
  Step 1: $800 - $200 = $600 remainder
  Step 2: $600 × 50% = $300 deduction
  Step 3: $600 - $300 = $300 countable

Parent 2: $400 earned
  Step 1: $400 - $200 = $200 remainder
  Step 2: $200 × 50% = $100 deduction
  Step 3: $200 - $100 = $100 countable

Total Countable = $300 + $100 = $400
Payment Standard (size 4, 2024) = $707
Benefit = $707 - $400 = $307/month
```

Source: [BEM 520 Section F](https://mdhhs-pres-prod.michigan.gov/olmweb/EX/BP/Public/BEM/520.pdf)

---

### 5. RESOURCE LIMITS

| Period | Resource Limit |
|--------|----------------|
| 2008-10-01 | $3,000 |
| 2019-01-01 | $15,000 |

Source: [BEM 400 (p.1)](https://mdhhs-pres-prod.michigan.gov/olmweb/EX/BP/Public/BEM/400.pdf#page=1)

---

## Variables Implemented

- `mi_fip` - Main benefit amount
- `mi_fip_eligible` - Overall eligibility
- `mi_fip_income_eligible` - Income eligibility (dual-test)
- `mi_fip_resources_eligible` - Resource eligibility
- `mi_fip_payment_standard` - Payment standard by family size
- `mi_fip_countable_income_for_eligibility` - Countable income (20% rate)
- `mi_fip_countable_income_for_benefit` - Countable income (50% rate)
- `mi_fip_countable_earned_income_for_eligibility` - Earned income after 20% deduction
- `mi_fip_countable_earned_income_for_benefit` - Earned income after 50% deduction
- `mi_fip_earned_income_after_deductions_for_eligibility_person` - Per-person 20% deduction
- `mi_fip_earned_income_after_deductions_for_benefit_person` - Per-person 50% deduction

## Test Coverage

- 66 unit and integration tests covering all variables

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: policyengine_us/parameters/gov/states/mi/mdhhs/fip/payment_standard/additional_person_increment.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  period: month
+  label: Michigan FIP additional person increment
+  reference:
+    - title: Michigan TANF State Plan 2017 - Payment Standards (frozen 2008-2024)
+      href: https://www.michigan.gov/-/media/Project/Websites/mdhhs/Folder1/Folder33/TANF_State_Plan_2017.pdf#page=7
+    - title: Michigan TANF State Plan Effective 01-01-23 Revised 07-01-25 - Updated Payment Standards
+      href: https://www.michigan.gov/mdhhs/-/media/Project/Websites/mdhhs/Inside-MDHHS/Reports-and-Statistics---Human-Services/State-Plans-and-Federal-Regulations/TANF_State_Plan_Effective_01-01-23_Rev-07-01-25.pdf
+    - title: Michigan DHHS Reference Table 210 - FIP Monthly Assistance Payment Standard
+      href: https://mdhhs-pres-prod.michigan.gov/olmweb/ex/RF/Public/RFT/210.pdf
+      publication_date: 2025-01-01
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/mdhhs/fip/payment_standard/additional_person_increment.yaml]: Lets remove publication dates

File: policyengine_us/variables/gov/states/mi/mdhhs/fip/mi_fip_payment_standard.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.mi.mdhhs.fip.payment_standard
+
+        # Determine household size
+        size = spm_unit("spm_unit_size", period)
+
+        # For sizes 1-7, use the bracket schedule
+        # For size 8+, use the size 7 amount plus additional per person
+        max_bracket_size = 7  # Maximum size in bracket schedule
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/mdhhs/fip/mi_fip_payment_standard.py]:
```suggestion
        max_household_size = 7  # Maximum size in bracket schedule
```

File: policyengine_us/variables/gov/states/mi/mdhhs/fip/mi_fip_payment_standard.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        size_capped = min_(size, max_bracket_size)
+        base_standard = p.amounts.calc(size_capped)
+
+        # Add additional amount for household size 8+
+        additional_persons = max_(size - max_bracket_size, 0)
+        additional_amount = additional_persons * p.additional_person_increment
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/mdhhs/fip/mi_fip_payment_standard.py]: Can we make the additional amount a separate variable?

File: policyengine_us/variables/gov/states/mi/mdhhs/fip/eligibility/mi_fip_eligible.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        # Must meet demographic requirements (age, deprivation)
+        # Use federal demographic eligibility
+        demographic_eligible = spm_unit("is_demographic_tanf_eligible", period)
+
+        # Must have at least one citizen or legal immigrant
+        # Per MI FIP requirements: "You must have at least one citizen or
+        # qualified lawful immigrant in your family"
+        has_citizen = spm_unit.any(
+            person("is_citizen_or_legal_immigrant", period)
+        )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/mdhhs/fip/eligibility/mi_fip_eligible.py]: If this is specifically stated in the regs. lets make this a separate list parameter with the immigration statuses

File: policyengine_us/parameters/gov/states/mi/mdhhs/fip/payment_standard/additional_person_increment.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+description: Michigan adds this amount to the Family Independence Program payment standard for each person beyond household size 7.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mi/mdhhs/fip/payment_standard/additional_person_increment.yaml]:
```suggestion
description: Michigan adds this amount to the Family Independence Program payment standard for each person beyond the maximum household size.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/mi/mdhhs/fip/eligibility/mi_fip_eligible.py (hunk 4 above)
Comment (hua7450) [reply policyengine_us/variables/gov/states/mi/mdhhs/fip/eligibility/mi_fip_eligible.py]: They are the exact same list.

**Comment by PavelMakarchuk** [comment]
## 📋 Automated PR Review Summary

### ✅ Strengths
- **Well-structured implementation** with proper separation between eligibility and benefit calculation
- **Correct variable naming** - all variables use the  prefix consistently
- **No hard-coded policy values** - all amounts, rates, and thresholds properly reference parameters
- **Good use of vectorization patterns** - proper use of `max_()`, `where()`, and avoids scalar logic
- **Comprehensive integration tests** - 15+ test cases covering various household scenarios
- **100% code coverage** as reported by Codecov
- **All CI checks passing**

### 🔍 Issues Found

#### Critical (Must Address)
- [ ] **4 missing unit test files**: No unit tests exist for:
  - `mi_fip_countable_income_for_benefit`
  - `mi_fip_countable_income_for_eligibility`
  - `mi_fip_countable_earned_income_for_benefit`
  - `mi_fip_countable_earned_income_for_eligibility`

#### Suggestions (Should Consider)

**From Existing Review Comments (Unresolved):**
- [ ] Remove publication dates from reference titles
- [ ] Make `additional` amount calculation a separate variable for clarity
- [ ] Consider making citizenship/immigration statuses a separate list parameter if specified in regulations

**From Automated Validation:**

**Federal/State Jurisdiction:**
- [ ] Review whether `tanf_gross_earned_income` and `tanf_gross_unearned_income` suffice for Michigan, or if Michigan-specific income variables are needed per BEM 502/503

**Documentation:**
- [ ] Add `documentation` attributes to variables explaining calculation methodology
- [ ] Consider adding a README.md documenting known limitations (time limits, work requirements, recoupment not modeled)

**Reference Quality:**
- [ ] Some reference URLs return 403 errors (TANF State Plan links)
- [ ] Reference titles should include specific quoted values for traceability (e.g., "$200 flat disregard")
- [ ] Parameters with multiple time-period values should have separate references for each effective date

**Test Coverage Gaps:**
- [ ] Missing boundary tests for income eligibility (exact threshold values)
- [ ] Missing historical period tests (pre-2024 parameter values, pre-2011 ongoing_rate)
- [ ] Missing enrollment status comparison tests (new applicant 20% vs enrolled 50%)
- [ ] Missing mixed citizen/non-citizen household tests

### 📊 Validation Results
- **Variable Naming**: ✅ PASS (0 issues)
- **Hard-Coded Values**: ✅ PASS (0 issues)
- **Vectorization**: ✅ PASS (0 issues)
- **Federal/State Jurisdiction**: ⚠️ 2 items to review
- **Documentation**: ⚠️ 4 items to review
- **Reference Quality**: ⚠️ 6 items to review
- **Test Coverage**: ⚠️ 4 missing test files, ~10 edge case gaps

### 🚀 Overall Assessment

This is a **solid implementation** of Michigan's Family Independence Program. The code follows PolicyEngine patterns correctly, uses proper parameterization, and has good integration test coverage. The main areas for improvement are:

1. Adding the 4 missing unit test files
2. Addressing the existing review comments from @PavelMakarchuk
3. Improving reference documentation quality

**Recommendation**: Address the critical items above, then this PR should be ready for merge.

---
🤖 *Automated review generated with Claude Code*

**Comment by PavelMakarchuk** [comment]
@hua7450 lets add the missing unit tests and merge

**Review by PavelMakarchuk — APPROVED** [review]
